.gitignore

```diff
@@ -2,6 +2,7 @@ extension-dist
 web-ext-artifacts
 todo.md
 .vscode
+build/
```

Comment:
Why do you need that? You didn't change anything in the build tooling, so there should not be a build folder

Reply:
I indeed didn't change the build tooling, but I did initially struggle to get the build working - at some point, it generated a `build/` folder, which I assumed probably should not be committed.

Reply:
The biggest stumbling block for me initially was that it seems to need a very specific version of Node to work; I recall versions in both Node 12 and Node 16 not working. 

Node 12 doesn't like TypeScript;

```
[!] SyntaxError: Unexpected token '.'
/Users/groenroos/Repositories/NebulaEnhance/rollup.config.js:259
  const type = args.configType?.toLowerCase();
                               ^
```

Node 16.10 has some other problems:

```
Assertion failed: (thread_id_key != 0x7777), function find_thread_id_key, file ../src/coroutine.cc, line 134.
zsh: abort
```

I only managed to get it working with Node 14.x - which is fine, but would be good to at least document the version requirement in the development docs.

I also was never able to get the Firefox command running on my computer (macOS 11.4);

```
s: The request "dev" profile name cannot be resolved to a profile path
```

For this reason, I've only been able to test my changes on Chromium.

Reply:
Also, the `In depth` section in `DEVELOPING.md` mentions a `lint-fix` command which doesn't exist; and the `lint` command fixes problems automatically, which you wouldn't expect (at least based on the docs).

Reply:
I use Node version v15.14.0, that's very interesting.

> Node 12 doesn't like TypeScript;
This problem isn't TypeScript, the config is in ES; but it seems I am using some newer syntax, didn't catch that; will document.

> Node 16.10 has some other problems:
That, frankly, looks like a bug in node, that's an internal assertion, no clue how you got that.

> I also was never able to get the Firefox command running on my compute
I'm very sorry about that, it seems I forgot about this. In my setup, to make sure my settings (and passwords) aren't lost when I restart, I have a dedicated dev profile, I didn't notice that this is a problem, thank you for pointing it out!

Reply:
> Also, the In depth section in DEVELOPING.md mentions a lint-fix command which doesn't exist; and the lint command fixes problems automatically, which you wouldn't expect (at least based on the docs).

You're not up-to-date ;) I changed that recently

Reply:
I haven't been able to re-create the `build/` folder, but it only contained a `config.gypi` as per [this](https://stackoverflow.com/questions/18023501/what-is-this-build-config-gypi-file-and-can-i-delete-it). I recall having some kind of problems with `node-gyp` when setting up - perhaps related to installing `pnpm`, or to the various Node version issues I was troubleshooting before I thought to try Node 14?

Reply:
Hmm, I'm not sure, I don't see `node-gyp` anywhere in the lock file, so it's probably not anywhere in the dependencies...

Very strange, might be related to the error on Node 16